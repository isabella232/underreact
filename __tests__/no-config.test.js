'use strict';

const path = require('path');
const tempy = require('tempy');
const fs = require('fs');

const hijackDel = require('./test-utils/hijack-del');
const removePath = require('./test-utils/remove-path');
const bootstrapReactRepo = require('./test-utils/bootstrap-react-repo');
const dirTree = require('directory-tree');

// The default behavior of `del` module doesn't allow modifying
// of files outside <rootDir>, which is something we want for the users
// but not in testing as the temp directory is outside.
// `hijackDel` modifies `del` to override this behavior for testing.
jest.mock('del', hijackDel);

const { copy } = require('../lib/packageable/auto-copy');
const config = require('../lib/config');
const build = require('../commands/build');

jest.setTimeout(15 * 1000);

describe('No Config Test', () => {
  const tempDir = tempy.directory();

  // give `urc.rootDirectory` a stable value
  const dirPath = path.join(tempDir, 'test');

  fs.mkdirSync(dirPath);

  beforeAll(async () => {
    await bootstrapReactRepo(dirPath);
    await copy({
      sourceDir: path.join(__dirname, 'fixtures', 'no-config'),
      destDir: dirPath
    });
  });

  test('builds the correct output', async () => {
    const buildOutput = await build(
      await config('build', {
        config: path.join(dirPath, 'underreact.config.js')
      })
    );
    expect(buildOutput).toBe(true);

    let tree = dirTree(dirPath);

    // Makes the snapshot stable by replacing the random dirPath
    // with the string `<TEMP_DIR>`
    tree = removePath({
      object: dirTree(dirPath),
      path: dirPath,
      replaceWith: '<TEMP_DIR>'
    });

    expect(tree).toMatchSnapshot();
  });
});
