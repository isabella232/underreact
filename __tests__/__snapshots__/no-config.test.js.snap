// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`No Config Test builds the correct output 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "extension": ".html",
          "name": "index.html",
          "path": "<TEMP_DIR>/_underreact-site/index.html",
          "size": 1904,
          "type": "file",
        },
        Object {
          "children": Array [
            Object {
              "extension": ".js",
              "name": "main-9d92f92f30.chunk.js",
              "path": "<TEMP_DIR>/_underreact-site/underreact-assets/main-9d92f92f30.chunk.js",
              "size": 1297,
              "type": "file",
            },
            Object {
              "extension": ".js",
              "name": "runtime-912ed6ecb8.js",
              "path": "<TEMP_DIR>/_underreact-site/underreact-assets/runtime-912ed6ecb8.js",
              "size": 1468,
              "type": "file",
            },
            Object {
              "extension": ".js",
              "name": "vendor-b00bbe6085.chunk.js",
              "path": "<TEMP_DIR>/_underreact-site/underreact-assets/vendor-b00bbe6085.chunk.js",
              "size": 175,
              "type": "file",
            },
          ],
          "name": "underreact-assets",
          "path": "<TEMP_DIR>/_underreact-site/underreact-assets",
          "size": 2940,
          "type": "directory",
        },
      ],
      "name": "_underreact-site",
      "path": "<TEMP_DIR>/_underreact-site",
      "size": 4844,
      "type": "directory",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "extension": ".js",
              "name": "index.js",
              "path": "<TEMP_DIR>/node_modules/react/index.js",
              "size": 21,
              "type": "file",
            },
            Object {
              "extension": ".json",
              "name": "package.json",
              "path": "<TEMP_DIR>/node_modules/react/package.json",
              "size": 232,
              "type": "file",
            },
          ],
          "name": "react",
          "path": "<TEMP_DIR>/node_modules/react",
          "size": 253,
          "type": "directory",
        },
        Object {
          "children": Array [
            Object {
              "extension": ".js",
              "name": "index.js",
              "path": "<TEMP_DIR>/node_modules/react-dom/index.js",
              "size": 25,
              "type": "file",
            },
            Object {
              "extension": ".json",
              "name": "package.json",
              "path": "<TEMP_DIR>/node_modules/react-dom/package.json",
              "size": 236,
              "type": "file",
            },
          ],
          "name": "react-dom",
          "path": "<TEMP_DIR>/node_modules/react-dom",
          "size": 261,
          "type": "directory",
        },
      ],
      "name": "node_modules",
      "path": "<TEMP_DIR>/node_modules",
      "size": 514,
      "type": "directory",
    },
    Object {
      "children": Array [
        Object {
          "extension": ".js",
          "name": "app.js",
          "path": "<TEMP_DIR>/src/app.js",
          "size": 163,
          "type": "file",
        },
        Object {
          "extension": ".js",
          "name": "entry.js",
          "path": "<TEMP_DIR>/src/entry.js",
          "size": 145,
          "type": "file",
        },
      ],
      "name": "src",
      "path": "<TEMP_DIR>/src",
      "size": 308,
      "type": "directory",
    },
    Object {
      "extension": ".js",
      "name": "underreact.config.js",
      "path": "<TEMP_DIR>/underreact.config.js",
      "size": 43,
      "type": "file",
    },
  ],
  "name": "test",
  "path": "<TEMP_DIR>",
  "size": 5709,
  "type": "directory",
}
`;
